lib/instrumentation/generic-span.js

```diff
@@ -8,16 +8,24 @@ const TraceParent = require('traceparent')
 
 module.exports = GenericSpan
 
-function GenericSpan (agent, type, opts) {
+function GenericSpan (agent, ...args) {
+  const opts = typeof args[args.length - 1] === 'object'
+    ? (args.pop() || {})
+    : {}
+
   this._timer = new Timer(opts.timer, opts.startTime)
   this._context = TraceParent.startOrResume(opts.childOf, agent._conf) // _context is used by the OT bridge, and should unfortunately therefore be considered public
   this._agent = agent
   this._labels = null
 
   this.timestamp = this._timer.start
-  this.type = type || 'custom'
   this.ended = false
   this.sync = true
+
+  this.type = null
```

Comment:
We should probably keep the initial value as the string `custom`:

```suggestion
  this.type = null || 'custom'
```

Either that, or we somehow need to make sure that it gets set to `custom` in another way if no value is provided.

Reply:
I changed it to just use `'custom'` as the fallback in the span/transaction encoding functions.

Reply:
perfect 👍